naviguide.cc

```diff
@@ -285,30 +301,67 @@ header_write()
 static void
 data_write()
 {
-  nof_wp = waypt_count();
-  if (nof_wp) {
-    header_write();
-    waypt_disp_all(ng_waypt_rd);
-  } else {
-    nof_wp = route_waypt_count();
-    if (nof_wp) {
-      header_write();
-      route_disp_all(NULL, NULL, ng_waypt_rd);
-    }
+
+  switch (write_type) {
+    case wptdata:
+      waypt_disp_all(ng_waypt_wr);
+      break;
+    case rtedata:
+      rte_head = route_head_alloc();
+      route_add_head(rte_head);
+      route_disp_all(NULL, NULL, ng_waypt_wr);
+      break;
+    case trkdata:
+      rte_head = route_head_alloc();
+      track_add_head(rte_head);
+      track_disp_all(NULL, NULL, ng_waypt_wr);
+      break;
+    default:
+      break;
   }
 }
 
 
 static void
 wr_init(const QString& fname)
 {
-  file_out = gbfopen_le(fname, "wb", MYNAME);
-  ng_fill_header_default();
-  if (NULL != reorder)
-    if (!case_ignore_strcmp(reorder, "y")) {
-      reorder_wp = 1;
+  int i;
+
+  for (i=0; write_as[i] && !nof_wp && i<3; i++) {
```

Comment:
"for (int i = 0; ..."
It'd be way less fragile if we could just write everything. Will that work?